src/auth/models/optionsMapper.class.ts

```diff
@@ -0,0 +1,40 @@
+import { AuthOptions } from './authOptions.interface'
+import { OpenIDMetadata } from '../oidc'
+import { ResponseType, ClientAuthMethod } from 'openid-client'
+import { OAuth2Metadata } from '../oauth2'
+
+export class OptionsMapper {
+    /* eslint-disable @typescript-eslint/camelcase */
+    public static getOpenIDOptions(openIdOptions: AuthOptions): OpenIDMetadata {
+        const options = {
+            client_id: openIdOptions.clientID,
+            client_secret: openIdOptions.clientSecret,
+            discovery_endpoint: openIdOptions.discoveryEndpoint,
+            issuer_url: openIdOptions.issuerUrl,
+            logout_url: openIdOptions.logoutUrl,
+            redirect_uri: openIdOptions.callbackURL,
+            response_types: openIdOptions.responseTypes as ResponseType[],
+            scope: openIdOptions.scope,
+            sessionKey: openIdOptions.sessionKey,
+            token_endpoint_auth_method: openIdOptions.tokenEndpointAuthMethod as ClientAuthMethod,
+            useRoutes: false,
+        }
+        /* eslint-disable @typescript-eslint/camelcase */
+        return options
+    }
+
+    public static getOAuth2Options(oAuth2Options: AuthOptions): OAuth2Metadata {
```

Comment:
I think, we don't actually need this one, the options going in are the same as the ones coming out

Reply:
Agreed. Remove it